Tests/AsyncHTTPClientTests/HTTPClientRequestTests.swift

```diff
@@ -0,0 +1,444 @@
+                .asAsyncSequence()
+                .map { ByteBuffer($0) }
+
+            request.body = .stream(asyncSequence)
+            var preparedRequest: PreparedRequest?
+            XCTAssertNoThrow(preparedRequest = try PreparedRequest(request))
+            guard let preparedRequest = preparedRequest else { return }
+
+            XCTAssertEqual(preparedRequest.poolKey, .init(
+                scheme: .http,
+                connectionTarget: .domain(name: "example.com", port: 80),
+                tlsConfiguration: nil
+            ))
+            XCTAssertEqual(preparedRequest.head, .init(
+                version: .http1_1,
+                method: .POST,
+                uri: "/post",
+                headers: [
+                    "host": "example.com",
+                    "transfer-encoding": "chunked",
+                ]
+            ))
+            XCTAssertEqual(preparedRequest.requestFramingMetadata, .init(
+                connectionClose: false,
+                body: .stream
+            ))
+            guard let buffer = await XCTAssertNoThrowWithResult(try await preparedRequest.body.read()) else { return }
+            XCTAssertEqual(buffer, .init(string: "post body"))
+        }
+        #endif
+    }
+
+    func testPostWithAsyncSequenceWithKnownLength() {
+        #if compiler(>=5.5) && canImport(_Concurrency)
+        guard #available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *) else { return }
+        XCTAsyncTest {
+            var request = Request(url: "http://example.com/post")
+            request.method = .POST
+            let asyncSequence = ByteBuffer(string: "post body")
+                .readableBytesView
+                .chunked(maxChunkSize: 2)
+                .asAsyncSequence()
+                .map { ByteBuffer($0) }
+
+            request.body = .stream(length: 9, asyncSequence)
+            var preparedRequest: PreparedRequest?
+            XCTAssertNoThrow(preparedRequest = try PreparedRequest(request))
+            guard let preparedRequest = preparedRequest else { return }
+
+            XCTAssertEqual(preparedRequest.poolKey, .init(
+                scheme: .http,
+                connectionTarget: .domain(name: "example.com", port: 80),
+                tlsConfiguration: nil
+            ))
+            XCTAssertEqual(preparedRequest.head, .init(
+                version: .http1_1,
+                method: .POST,
+                uri: "/post",
+                headers: [
+                    "host": "example.com",
+                    "content-length": "9",
+                ]
+            ))
+            XCTAssertEqual(preparedRequest.requestFramingMetadata, .init(
+                connectionClose: false,
+                body: .fixedSize(9)
+            ))
+            guard let buffer = await XCTAssertNoThrowWithResult(try await preparedRequest.body.read()) else { return }
+            XCTAssertEqual(buffer, .init(string: "post body"))
+        }
+        #endif
+    }
```

Comment:
Those are awesome! Thanks! The only other thing I can think of might, be three tests for the different unix schemes.

Reply:
Oh and we should have an explicit test on the ConnectionPool.Key ensuring that `unix+http` != `unix`